base_location/views/better_zip_view.xml

```diff
@@ -7,11 +7,23 @@
         <field name="arch" type="xml">
             <form string="City">
                 <group col="4">
-                <field name="name"/>
-                <field name="code"/>
-                <field name="city"/>
-                <field name="state_id"/>
-                <field name="country_id"/>
+                    <group>
+                        <field name="name"/>
+                        <field name="city_id"
+                               attrs="{
+                                'invisible': [('enforce_cities', '=', False)],
+                                'required': [('enforce_cities', '=', True)]}"
+                               domain="[('country_id', '=', country_id)]"/>
```

Comment:
This should be dynamically added on country_id onchange, as this way, no city is shown if no country is selected.